ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKStream.java

```diff
@@ -438,4 +532,41 @@ public FunctionRegistry getFunctionRegistry() {
   public SchemaRegistryClient getSchemaRegistryClient() {
     return schemaRegistryClient;
   }
+
+  protected static class KsqlValueJoiner
```

Comment:
We should try and test this class.

Reply:
There is a unit test for it for each (nonNull, nonNull), (nonNull, null), and (null, nonNull) scenario.